snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -227,14 +222,21 @@ def mark_cleaned(self, step):
             os.rmdir(self.statedir)
 
     def _fetch_stage_packages(self):
-        try:
-            self.stage_packages = self._stage_package_handler.fetch()
-        except repo.errors.PackageNotFoundError as e:
-            raise RuntimeError("Error downloading stage packages for part "
-                               "{!r}: {}".format(self.name, e.message))
+        stage_packages = self._grammar_processor.get_stage_packages()
```

Comment:
I would of loved to have this processed already by the `project_loader` but I guess that is hard due to the repo sources thing :-/